conf.go

```diff
@@ -0,0 +1,98 @@
+package hdfs
+
+import (
+	"encoding/xml"
+	"errors"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path"
+	"strings"
+)
+
+// Property is the struct representation of hadoop configuration
+// key value pair.
+type Property struct {
+	Name  string `xml:"name"`
+	Value string `xml:"value"`
+}
+
+type propertyList struct {
+	Property []Property `xml:"property"`
+}
+
+// HadoopConf represents a map of all the key value configutation
+// pairs found in a user's hadoop configuration files.
+type HadoopConf map[string]string
+
+var errUnresolvedNamenode = errors.New("no namenode address in configuration")
+
+// LoadHadoopConf returns a HadoopConf object that is key value map
+// of all the hadoop conf properties, swallows errors reading xml
+// and reading a non-existant file.
+func LoadHadoopConf(inputPath string) HadoopConf {
+	var tryPaths []string
+
+	if inputPath != "" {
+		tryPaths = append(tryPaths, inputPath)
+	} else {
+		hadoopConfDir := os.Getenv("HADOOP_CONF_DIR")
+		hadoopHome := os.Getenv("HADOOP_HOME")
+		if hadoopConfDir != "" {
+			confHdfsPath := path.Join(hadoopConfDir, "hdfs-site.xml")
+			confCorePath := path.Join(hadoopConfDir, "core-site.xml")
+			tryPaths = append(tryPaths, []string{confHdfsPath, confCorePath}...)
+		}
+		if hadoopHome != "" {
+			hdfsPath := path.Join(hadoopHome, "conf", "hdfs-site.xml")
```

Comment:
I think you want `filepath.Join` instead of `path.Join`, since it's on the local filesystem.